src/utilities/units.ts

```diff
@@ -43,6 +43,88 @@ export function toFixed(val: number, decimals: number = 0): string {
     return String(val);
 }
 
+export function trimTrailingDecimals(value: string): string {
+    let splitValue: string[] = value.split(".");
+
+    if (splitValue[1] === undefined) {
+        return splitValue[0];
+    } else {
+        for (let i = 0; i < splitValue[1].length; i++) {
+            if (splitValue[1][i] !== "0") {
+                return value;
+            }
+        }
+        return splitValue[0];
+    }
+}
+
+export function trimTrailingZeros(value: string): string {
+    // Trims the trailing zeros from the input decimal value. If
+    // all trailing values are '0', we return only the values
+    // left of the decimal
+
+    var decimals = value.split(".");
+    var trimmed = decimals[1];
+    var temp = [];
+
+    if (typeof decimals[1] != "undefined") {
+        temp = decimals[1].split("");
+    } else {
+        return decimals[0];
+    }
+
+    for (var i = decimals[1].length - 1; i >= 0; i--) {
+        // Check if trailing value is 0 and pop() value if so.
+        if (decimals[1][i] === "0") {
+            temp.pop();
+            trimmed = temp.join("");
+        } else {
+            // Once all trailing values are removed, rebuild full value minus
+            // trailing zeros and return.
+
+            trimmed = decimals[0] + "." + trimmed;
+            return trimmed;
+        }
+    }
+    return decimals[0];
+}
+
+export function getVariablePrecision(value: number): number {
+    // Estimates the precision of input tick value. Input provides
+    // delta between neighboring tick values and iterates through
+    // up to 14 decimal places to determine the approxmiate
+    // precision.
+
+    var decimalPlacement = 0.1;
+    var precision = 3;
+
+    for (var i = 0; i < 9; i++) {
+        if (value < decimalPlacement) {
+            decimalPlacement = 0.1 * decimalPlacement;
+            precision++;
+        } else {
+            return precision;
+        }
+    }
+
+    return precision;
+}
+
+export function toFormattedNotation(value: number, delta: number): string {
```

Comment:
We need a better naming to distinguish `toFormattedNotation` and `formattedNotation`. Perhaps `formattedNotationByDiff`?

Reply:
I would like to suggest a best practices document for these things. If there was a documented naming convention it would have saved half the changes requests. The purpose of naming it `toFormattedNotation` was to match the standard `toFixed` format, which I think is actually better than `formattedNotation`. I had intended to remove the later if my function was ok'd.

I'll make all these changes on Monday when I am back in the office.